velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -218,11 +230,11 @@ void Writer::write(const VectorPtr& data) {
       auto recordBatch, arrow::ImportRecordBatch(&array, &schema));
   if (!arrowContext_->schema) {
     arrowContext_->schema = recordBatch->schema();
-    for (int colIdx = 0; colIdx < arrowContext_->schema->num_fields();
-         colIdx++) {
-      arrowContext_->stagingChunks.push_back(
-          std::vector<std::shared_ptr<arrow::Array>>());
-    }
+  }
+
+  for (int colIdx = 0; colIdx < arrowContext_->schema->num_fields(); colIdx++) {
```

Comment:
Every `write` call will incur this overhead.
It should be okay to duplicate this loop above in the constructor.

Reply:
Done.